server/index.js

```diff
@@ -17,12 +17,40 @@ app.use(express.json());
 app.get('/api/products/', (req, res, next) => {
   const sqlGet = 'select "productId", "name", "price", "image", "shortDescription" from "products" ';
   db.query(sqlGet)
-  
+
     .then(result => res.json(result.rows)) // drop the array index so the whole array will req/res
 
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const parsedID = parseInt(req.params.productId);
+  if (isNaN(parsedID)) {
+    next(new ClientError('invalid data type', 400));
```

Comment:
Ok, you just need to put a `return` here to prevent the rest of the code from running.

Reply:
I got the return statement in there.